### ISSUE TITLE
Filter woocommerce_checkout_customer_id is not available during blocks checkout

### ISSUE THREAD

**Issue by webdados** [woothemes/woocommerce#44530]
### Prerequisites

- [ ] I have carried out troubleshooting steps and I believe I have found a bug.
- [ ] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

On the classic checkout, the `woocommerce_checkout_customer_id` filter allows attributing the order to another user, which is not the current logged-in one:
https://github.com/woocommerce/woocommerce/blob/8.6.0-rc.1/plugins/woocommerce/includes/class-wc-checkout.php#L435
https://github.com/woocommerce/woocommerce/blob/8.6.0-rc.1/plugins/woocommerce/includes/class-wc-checkout.php#L1114

On the blocks checkout this filter is no longer used:
https://github.com/woocommerce/woocommerce/blob/8.6.0-rc.1/plugins/woocommerce/src/StoreApi/Utilities/OrderController.php#L111
https://github.com/woocommerce/woocommerce/blob/8.6.0-rc.1/plugins/woocommerce/src/StoreApi/Routes/V1/Checkout.php#L515

### Expected behavior

The blocks checkout should allow for the order customer to be changed by 3rd parties.

### Actual behavior

The blocks checkout does not allow to change the order user id via filters.

### Steps to reproduce

Add this to your functions.php file:
`add_filter( 'woocommerce_checkout_customer_id', function( $user_id ) {
	return 22;
} );`

Shop using the classic and blocks checkout.

### WordPress Environment

`
### WordPress Environment ###

WordPress address (URL): https://wordpress.local
Site address (URL): https://wordpress.local
WC Version: 8.6.0
REST API Version: ✔ 8.6.0
WC Blocks Version: ✔ 11.8.0-dev
Action Scheduler Version: ✔ 3.7.1
Log Directory Writable: ✔
WP Version: ❌ 6.5-alpha-57571 . Há uma nova versão do WordPress disponível (6.5-alpha-57580)
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: ✔
WP Cron: –
Language: pt_PT
External object cache: –

### Server Environment ###

Server Info: nginx/1.16.0
PHP Version: 8.1.9
PHP Post Max Size: 1.000 MB
PHP Time Limit: 1200
PHP Max Input Vars: 4000
cURL Version: 7.84.0
(SecureTransport) OpenSSL/1.1.1q

SUHOSIN Installed: –
MySQL Version: 8.0.16
Max Upload Size: 300 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 8.6.0
WC Database Prefix: wp_
Tamanho total da base de dados: 3841.97MB
Tamanho dos dados da base de dados: 3784.07MB
Tamanho do índice da base de dados: 57.90MB
wp_woocommerce_sessions: Dados: 1.45MB + Índice: 0.06MB + Engine InnoDB
wp_woocommerce_api_keys: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: Dados: 0.02MB + Índice: 0.06MB + Engine InnoDB
wp_woocommerce_order_items: Dados: 0.19MB + Índice: 0.08MB + Engine InnoDB
wp_woocommerce_order_itemmeta: Dados: 1.52MB + Índice: 1.84MB + Engine InnoDB
wp_woocommerce_tax_rates: Dados: 0.02MB + Índice: 0.06MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: Dados: 0.02MB + Índice: 0.05MB + Engine InnoDB
wp_woocommerce_shipping_zones: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: Dados: 0.02MB + Índice: 0.05MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_woocommerce_log: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_actionscheduler_actions: Dados: 2.02MB + Índice: 0.73MB + Engine InnoDB
wp_actionscheduler_claims: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_actionscheduler_groups: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_actionscheduler_logs: Dados: 1.50MB + Índice: 0.44MB + Engine InnoDB
wp_aelia_dismissed_messages: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_aelia_exchange_rates_history: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_arm_activity: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_arm_bank_transfer_log: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_arm_email_templates: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_arm_entries: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_arm_fail_attempts: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_arm_form_field: Dados: 0.06MB + Índice: 0.00MB + Engine InnoDB
wp_arm_forms: Dados: 0.13MB + Índice: 0.00MB + Engine InnoDB
wp_arm_lockdown: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_arm_login_history: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_arm_member_templates: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_arm_members: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_arm_membership_setup: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_arm_payment_log: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_arm_subscription_plans: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_arm_termmeta: Dados: 0.06MB + Índice: 0.06MB + Engine InnoDB
wp_automatewoo_abandoned_carts: Dados: 0.02MB + Índice: 0.08MB + Engine InnoDB
wp_automatewoo_customer_meta: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_automatewoo_customers: Dados: 0.02MB + Índice: 0.09MB + Engine InnoDB
wp_automatewoo_events: Dados: 0.02MB + Índice: 0.05MB + Engine InnoDB
wp_automatewoo_guest_meta: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_automatewoo_guests: Dados: 0.02MB + Índice: 0.06MB + Engine InnoDB
wp_automatewoo_log_meta: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_automatewoo_logs: Dados: 0.02MB + Índice: 0.05MB + Engine InnoDB
wp_automatewoo_queue: Dados: 0.02MB + Índice: 0.05MB + Engine InnoDB
wp_automatewoo_queue_meta: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_automatewoo_unsubscribes: Dados: 0.02MB + Índice: 0.05MB + Engine InnoDB
wp_berocket_termmeta: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_bwf_contact: Dados: 0.02MB + Índice: 0.06MB + Engine InnoDB
wp_bwf_contact_meta: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_bwf_conversion_tracking: Dados: 0.02MB + Índice: 0.05MB + Engine InnoDB
wp_bwf_funnelmeta: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_bwf_funnels: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_bwf_optin_entries: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_bwf_wc_customers: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_chefpanda_restrictions: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_commentmeta: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_comments: Dados: 1.52MB + Índice: 0.48MB + Engine InnoDB
wp_CTT: Dados: 0.00MB + Índice: 0.00MB + Engine MyISAM
wp_CTT_marketplace_order: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_egoi_form_subscribers: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_egoi_map_fields: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_et_bloom_stats: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_eupago_mbway: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_ewwwio_images: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_facetwp_cache: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_facetwp_index: Dados: 0.02MB + Índice: 0.05MB + Engine InnoDB
wp_facetwp_temp: Dados: 2.52MB + Índice: 4.45MB + Engine InnoDB
wp_favethemes_currency_converter: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_fcc_currencies: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_font_color: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_frm_fields: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_frm_forms: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_frm_item_metas: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_frm_items: Dados: 0.02MB + Índice: 0.08MB + Engine InnoDB
wp_gp_glossaries: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_gp_glossary_entries: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_gp_meta: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_gp_originals: Dados: 0.02MB + Índice: 0.05MB + Engine InnoDB
wp_gp_permissions: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_gp_projects: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_gp_translation_sets: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_gp_translations: Dados: 0.02MB + Índice: 0.11MB + Engine InnoDB
wp_homey_earnings: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_homey_fields_builder: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_homey_map: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_homey_payouts: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_homey_thread_messages: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_homey_threads: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_ibwp_mp_submission: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_icl_background_task: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_icl_content_status: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_icl_core_status: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_icl_flags: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_icl_languages: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_icl_languages_translations: Dados: 0.19MB + Índice: 0.11MB + Engine InnoDB
wp_icl_locale_map: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_icl_message_status: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_icl_mo_files_domains: Dados: 0.05MB + Índice: 0.02MB + Engine InnoDB
wp_icl_node: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_icl_reminders: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_icl_string_batches: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_icl_string_packages: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_icl_string_pages: Dados: 0.27MB + Índice: 0.39MB + Engine InnoDB
wp_icl_string_positions: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_icl_string_status: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_icl_string_translations: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_icl_string_urls: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_icl_strings: Dados: 0.19MB + Índice: 0.38MB + Engine InnoDB
wp_icl_translate: Dados: 0.08MB + Índice: 0.02MB + Engine InnoDB
wp_icl_translate_job: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_icl_translation_batches: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_icl_translation_downloads: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_icl_translation_status: Dados: 0.05MB + Índice: 0.02MB + Engine InnoDB
wp_icl_translations: Dados: 1.52MB + Índice: 3.77MB + Engine InnoDB
wp_itsec_lockouts: Dados: 0.02MB + Índice: 0.08MB + Engine InnoDB
wp_itsec_log: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_itsec_temp: Dados: 0.02MB + Índice: 0.06MB + Engine InnoDB
wp_ix_log: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_jckwds: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_links: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_mipconnector_attribute_group_map: Dados: 0.02MB + Índice: 0.05MB + Engine InnoDB
wp_mipconnector_attribute_map: Dados: 0.02MB + Índice: 0.05MB + Engine InnoDB
wp_mipconnector_brand_map: Dados: 0.02MB + Índice: 0.05MB + Engine InnoDB
wp_mipconnector_brand_plugin_map: Dados: 0.02MB + Índice: 0.05MB + Engine InnoDB
wp_mipconnector_cache: Dados: 0.02MB + Índice: 0.05MB + Engine InnoDB
wp_mipconnector_category_map: Dados: 0.02MB + Índice: 0.05MB + Engine InnoDB
wp_mipconnector_file_log: Dados: 0.02MB + Índice: 0.06MB + Engine InnoDB
wp_mipconnector_image_url: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_mipconnector_import_process_attribute: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_mipconnector_import_process_attribute_group: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_mipconnector_import_process_brand: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_mipconnector_import_process_brand_plugin: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_mipconnector_import_process_category: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_mipconnector_import_process_product: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_mipconnector_import_process_tag: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_mipconnector_import_process_variation: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_mipconnector_order_log: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_mipconnector_product_image_url: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_mipconnector_product_map: Dados: 0.02MB + Índice: 0.05MB + Engine InnoDB
wp_mipconnector_product_url: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_mipconnector_reference_data: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_mipconnector_shipping_service: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_mipconnector_tag_map: Dados: 0.02MB + Índice: 0.09MB + Engine InnoDB
wp_mipconnector_variation_map: Dados: 0.02MB + Índice: 0.05MB + Engine InnoDB
wp_options: Dados: 3684.23MB + Índice: 0.33MB + Engine InnoDB
wp_pmxe_exports: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_pmxe_google_cats: Dados: 0.39MB + Índice: 0.00MB + Engine InnoDB
wp_pmxe_posts: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_pmxe_templates: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_pmxi_files: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_pmxi_history: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_pmxi_images: Dados: 0.05MB + Índice: 0.00MB + Engine InnoDB
wp_pmxi_imports: Dados: 0.28MB + Índice: 0.00MB + Engine InnoDB
wp_pmxi_posts: Dados: 1.52MB + Índice: 0.38MB + Engine InnoDB
wp_pmxi_templates: Dados: 0.14MB + Índice: 0.00MB + Engine InnoDB
wp_po_plugins: Dados: 0.02MB + Índice: 0.06MB + Engine InnoDB
wp_portuguese_postcodes: Dados: 23.55MB + Índice: 0.00MB + Engine InnoDB
wp_postmeta: Dados: 30.56MB + Índice: 24.06MB + Engine InnoDB
wp_posts: Dados: 5.52MB + Índice: 3.73MB + Engine InnoDB
wp_psmsc_agents: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_psmsc_attachments: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_psmsc_categories: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_psmsc_custom_fields: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_psmsc_customers: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_psmsc_email_notifications: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_psmsc_email_otp: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_psmsc_holidays: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_psmsc_logs: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_psmsc_options: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_psmsc_priorities: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_psmsc_statuses: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_psmsc_threads: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_psmsc_tickets: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_psmsc_wh_exceptions: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_psmsc_working_hrs: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_rank_math_404_logs: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_rank_math_internal_links: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_rank_math_internal_meta: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_rank_math_redirections: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_rank_math_redirections_cache: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_rank_math_sc_analytics: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_seur_custom_rates: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_seur_svpr: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_shiptimize: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_shiptimize_marketplace_order: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_sibs_payment_information: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_sibs_transaction: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_social_users: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_sp_admins: Dados: 0.00MB + Índice: 0.00MB + Engine MyISAM
wp_sp_answers: Dados: 0.00MB + Índice: 0.00MB + Engine MyISAM
wp_sp_config: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_sp_exceptions: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_sp_hooks_admins_multi: Dados: 0.00MB + Índice: 0.00MB + Engine MyISAM
wp_sp_hooks_customers: Dados: 0.00MB + Índice: 0.00MB + Engine MyISAM
wp_sp_hooks_templates: Dados: 0.00MB + Índice: 0.00MB + Engine MyISAM
wp_sp_hooks_unicode: Dados: 0.00MB + Índice: 0.00MB + Engine MyISAM
wp_sp_marketing_filter: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_sp_optout_deleted: Dados: 0.00MB + Índice: 0.00MB + Engine MyISAM
wp_sp_optout_prices: Dados: 0.00MB + Índice: 0.00MB + Engine MyISAM
wp_sp_optout_texts: Dados: 0.00MB + Índice: 0.00MB + Engine MyISAM
wp_sp_outofstock: Dados: 0.00MB + Índice: 0.00MB + Engine MyISAM
wp_sp_ownnumbersenderIDs: Dados: 0.00MB + Índice: 0.00MB + Engine MyISAM
wp_sp_routes: Dados: 0.00MB + Índice: 0.00MB + Engine MyISAM
wp_sp_settings: Dados: 0.00MB + Índice: 0.00MB + Engine MyISAM
wp_sp_sms_history: Dados: 0.00MB + Índice: 0.00MB + Engine MyISAM
wp_sp_smsuser: Dados: 0.00MB + Índice: 0.00MB + Engine MyISAM
wp_sp_template: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_sp_textsenderIDs: Dados: 0.00MB + Índice: 0.00MB + Engine MyISAM
wp_sti_atos_eleitorais: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_sti_distritos: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_sti_entidades: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_sti_locais: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_sti_temp_associados: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_sti_temp_associados_exist: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_term_relationships: Dados: 1.52MB + Índice: 1.52MB + Engine InnoDB
wp_term_taxonomy: Dados: 0.06MB + Índice: 0.06MB + Engine InnoDB
wp_termmeta: Dados: 0.09MB + Índice: 0.11MB + Engine InnoDB
wp_terms: Dados: 0.06MB + Índice: 0.03MB + Engine InnoDB
wp_tm_taskmeta: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_tm_tasks: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_usermeta: Dados: 0.13MB + Índice: 0.11MB + Engine InnoDB
wp_users: Dados: 0.02MB + Índice: 0.05MB + Engine InnoDB
wp_wc_admin_note_actions: Dados: 0.08MB + Índice: 0.02MB + Engine InnoDB
wp_wc_admin_notes: Dados: 0.09MB + Índice: 0.00MB + Engine InnoDB
wp_wc_appointment_relationships: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_wc_appointments_availability: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_wc_category_lookup: Dados: 0.05MB + Índice: 0.00MB + Engine InnoDB
wp_wc_customer_lookup: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_wc_deposits_payment_plans: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wc_deposits_payment_plans_schedule: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_wc_download_log: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_wc_ie_scheduled_docs: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wc_order_addresses: Dados: 0.19MB + Índice: 0.25MB + Engine InnoDB
wp_wc_order_coupon_lookup: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_wc_order_operational_data: Dados: 0.14MB + Índice: 0.06MB + Engine InnoDB
wp_wc_order_product_lookup: Dados: 0.08MB + Índice: 0.06MB + Engine InnoDB
wp_wc_order_stats: Dados: 0.08MB + Índice: 0.05MB + Engine InnoDB
wp_wc_order_tax_lookup: Dados: 0.05MB + Índice: 0.03MB + Engine InnoDB
wp_wc_orders: Dados: 0.30MB + Índice: 0.38MB + Engine InnoDB
wp_wc_orders_meta: Dados: 1.52MB + Índice: 2.94MB + Engine InnoDB
wp_wc_product_attributes_lookup: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_wc_product_download_directories: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_wc_product_meta_lookup: Dados: 1.52MB + Índice: 1.28MB + Engine InnoDB
wp_wc_rate_limits: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_wc_reserved_stock: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wc_revolut_orders: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_wc_tax_rate_classes: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_wc_webhooks: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_wcfm_daily_analysis: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_wcfm_detailed_analysis: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wcfm_enquiries: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wcfm_enquiries_meta: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wcfm_enquiries_response: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wcfm_enquiries_response_meta: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wcfm_fbc_chat_rows: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wcfm_fbc_chat_sessions: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_wcfm_fbc_chat_visitors: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wcfm_fbc_offline_messages: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wcfm_following_followers: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wcfm_marketplace_orders: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_wcfm_marketplace_orders_meta: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wcfm_marketplace_product_multivendor: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_wcfm_marketplace_refund_request: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wcfm_marketplace_refund_request_meta: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wcfm_marketplace_reverse_withdrawal: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wcfm_marketplace_reverse_withdrawal_meta: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wcfm_marketplace_review_rating_meta: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wcfm_marketplace_reviews: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wcfm_marketplace_reviews_response: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wcfm_marketplace_reviews_response_meta: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wcfm_marketplace_shipping_zone_locations: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wcfm_marketplace_shipping_zone_methods: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wcfm_marketplace_store_taxonomies: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wcfm_marketplace_vendor_ledger: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wcfm_marketplace_withdraw_request: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wcfm_marketplace_withdraw_request_meta: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wcfm_messages: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wcfm_messages_modifier: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wcfm_messages_stat: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wcfm_support: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wcfm_support_meta: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wcfm_support_response: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wcfm_support_response_meta: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wcpdf_invoice_number: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wdr_order_discounts: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wdr_order_item_discounts: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wdr_rules: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wfacp_stats: Dados: 0.02MB + Índice: 0.06MB + Engine InnoDB
wp_wfco_report_views: Dados: 0.02MB + Índice: 0.05MB + Engine InnoDB
wp_woo_dpd_pickup_points: Dados: 2.52MB + Índice: 0.27MB + Engine InnoDB
wp_woo_wallet_transaction_meta: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_woo_wallet_transactions: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_woocommerce_api_manager_secure_hash: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_woocommerce_oppcw_customer_contexts: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_woocommerce_oppcw_documents: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_woocommerce_oppcw_ecc: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_woocommerce_oppcw_schema_version: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_woocommerce_oppcw_storage: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_woocommerce_oppcw_transactions: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_table_rates: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wpbdp_fees: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wpbdp_form_fields: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_wpbdp_listing_fees: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_wpbdp_payments: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_wpbdp_payments_items: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wpbdp_submit_state: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wpmailsmtp_debug_events: Dados: 0.11MB + Índice: 0.00MB + Engine InnoDB
wp_wpmailsmtp_tasks_meta: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wpmudev_chat_log: Dados: 0.02MB + Índice: 0.06MB + Engine InnoDB
wp_wpmudev_chat_message: Dados: 0.02MB + Índice: 0.11MB + Engine InnoDB
wp_wpmudev_chat_users: Dados: 0.02MB + Índice: 0.06MB + Engine InnoDB
wp_wps_hit: Dados: 0.02MB + Índice: 0.14MB + Engine InnoDB
wp_wps_index: Dados: 0.02MB + Índice: 0.05MB + Engine InnoDB
wp_wps_key: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_wps_object_term: Dados: 0.02MB + Índice: 0.09MB + Engine InnoDB
wp_wps_object_type: Dados: 0.02MB + Índice: 0.08MB + Engine InnoDB
wp_wps_query: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_wps_uri: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_wps_user_agent: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_wpvivid_scan_result: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wpvivid_unused_uploads_files: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wsal_metadata: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_wsal_occurrences: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_yith_wcwl: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_yith_wcwl_lists: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_yoast_indexable: Dados: 10.52MB + Índice: 4.42MB + Engine InnoDB
wp_yoast_indexable_hierarchy: Dados: 0.02MB + Índice: 0.05MB + Engine InnoDB
wp_yoast_migrations: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_yoast_primary_term: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_yoast_seo_links: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_yoast_seo_meta: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB

### Post Type Counts ###

acf-field: 19
acf-field-group: 3
amn_smtp: 1
atividade: 4
attachment: 6283
aw_workflow: 2
blocks: 8
chefpanda_market: 3
easy-pricing-table: 1
featured_item: 8
frm_form_actions: 1
frm_styles: 1
homey_reservation: 2
ibwp_mp_popup: 1
listing: 1
mb_payment_request: 1
nav_menu_item: 27
oksofasstore: 3
page: 63
person: 5
post: 26
private_post: 4
product: 7958
product_variation: 131
revision: 227
shop_coupon: 20
shop_order: 691
shop_order_refund: 89
shop_subscription: 10
store: 1
szbdzones: 1
was: 1
wc_appointment: 1
wc_conditional_fee: 1
wc_order_status: 9
wc-donation: 2
wc-donation-report: 5
wcdp_payment: 4
wd_client_contract: 1
wd_support_ticket: 1
webdados_client_serv: 14
webdados_service: 8
wfacp_checkout: 1
wffn_ty: 1
wishlist: 1
wp_block: 1
wp_global_styles: 4
wp_navigation: 1
wp_template: 2
wpbdp_listing: 1
wpcf7_contact_form: 6
wps_subscriptions: 1
wpsc_ticket_thread: 3

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ❌As mensagens de erro não devem ser exibidas aos visitantes.

### Active Plugins (5) ###

WooCommerce Beta Tester: por WooCommerce – 2.2.4
WooCommerce: por Automattic – 8.6.0-rc.1
WordPress Beta Tester: por WordPress Upgrade/Install Team – 3.5.5

### Inactive Plugins (308) ###

Not relevant

### Dropin Plugins (1) ###

db.php: db.php

### Settings ###

API Enabled: –
Force SSL: –
Currency: EUR (€)
Currency Position: right_space
Thousand Separator: .
Decimal Separator: ,
Number of Decimals: 2
Taxonomies: Product Types: appointment (appointment)
external (external)
grouped (grouped)
mwb_booking (mwb_booking)
simple (simple)
subscription (subscription)
variable (variable)
variable subscription (variable-subscription)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to Woo.com: –
Enforce Approved Product Download Directories: –
HPOS feature screen enabled: ✔
HPOS feature enabled: ✔
Order datastore: Automattic\WooCommerce\Internal\DataStores\Orders\OrdersTableDataStore
HPOS data sync enabled: ✔

### WC Pages ###

Página da Loja: #14857 - /loja-pt/
Carrinho: #55 - /carrinho-de-compras/
Finalizar compras: #16786 - /checkout-2/
A minha conta: #16787 - /my-account-2/
Termos e condições: ❌ Página não definida

### Theme ###

Name: Storefront Child
Version: 20230130.01
Author URL: 
Child Theme: ✔
Parent Theme Name: Storefront
Parent Theme Version: 4.5.4
Parent Theme Author URL: https://woo.com/
WooCommerce Support: ✔

### Templates ###

Overrides: –

### Admin ###

Enabled Features: activity-panels
analytics
product-block-editor
coupons
core-profiler
customer-effort-score-tracks
import-products-task
experimental-fashion-sample-products
shipping-smart-defaults
shipping-setting-tour
homescreen
marketing
mobile-app-banner
navigation
onboarding
onboarding-tasks
product-variation-management
product-virtual-downloadable
product-external-affiliate
product-grouped
product-linked
remote-inbox-notifications
remote-free-extensions
payment-gateway-suggestions
shipping-label-banner
subscriptions
store-alerts
transient-notices
woo-mobile-welcome
wc-pay-promotion
wc-pay-welcome-page

Disabled Features: customize-store
minified-js
new-product-management-experience
product-pre-publish-modal
settings
async-product-editor-category-field
beta-tester-slotfill-examples

Daily Cron: ✔ Next scheduled: 2023-11-22 15:19:07 +00:00
Options: ✔
Notes: 175
Onboarding: completed

### Action Scheduler ###

Concluída: 138
Oldest: 2024-01-10 15:36:02 +0000
Newest: 2024-02-10 11:10:26 +0000

Falhada: 2.333
Oldest: 2023-02-01 17:31:32 +0000
Newest: 2024-02-07 23:32:07 +0000

Pendente: 2
Oldest: 2024-02-11 11:06:36 +0000
Newest: 2024-02-11 11:06:37 +0000

### Status report information ###

Generated at: 2024-02-10 11:29:27 +00:00
`

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woo.com/products/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

**Comment by webdados** [comment]
I've tried to add the filter to https://github.com/woocommerce/woocommerce/blob/8.6.0-rc.1/plugins/woocommerce/src/StoreApi/Utilities/OrderController.php#L111 on my dev install and it works. The order is assigned to the user returned to the `woocommerce_checkout_customer_id` filter.

`$order->set_customer_id( apply_filters( 'woocommerce_checkout_customer_id', get_current_user_id() ) );`

**Comment by wavvves** [comment]
Hi @webdados, thank for reporting this.
Indeed this filter seems to be left out from Store API. I added the issue to our backlog for planning.

**Comment by webdados** [comment]
> I've tried to add the filter to https://github.com/woocommerce/woocommerce/blob/8.6.0-rc.1/plugins/woocommerce/src/StoreApi/Utilities/OrderController.php#L111 on my dev install and it works. The order is assigned to the user returned to the `woocommerce_checkout_customer_id` filter.
> 
> `$order->set_customer_id( apply_filters( 'woocommerce_checkout_customer_id', get_current_user_id() ) );`

It's also on https://github.com/woocommerce/woocommerce/blob/8.6.0-rc.1/plugins/woocommerce/src/StoreApi/Routes/V1/Checkout.php#L515 but this seems to be called when a new customer checks-out and decides to create an account. Can you confirm this? In that case the filter shouldn't be necessary on this, only on the OrderController.php

**Comment by webdados** [comment]
Any plans on implementing this?

**Comment by wavvves** [comment]
Hi @webdados, this issue has not been selected yet for addressing. 
Although it is in our backlog since it is a feature from shortcode that has not been ported to blocks, it will greatly help prioritize if you can provide us with a use case 👍🏼

**Comment by webdados** [comment]
We have a plugin that allows shopping as a customer on the frontend and then attribute that order to a customer. It's for phone and email orders.
it works just fine on the classic checkout, but not on the blocks checkout because this is lacking.

**Comment by s3rgiosan** [comment]
Hi @wavvves 

For additional context, our current approach adds a field at the top of the checkout where the shop manager can search for and select a specific customer. When a customer is selected, we override the logged-in user’s data (the shop manager) with the selected customer’s data while storing the original logged-in user’s data in the session. After the order is processed, we restore the logged-in user’s data from the session.

This approach introduces several challenges, including managing various checkout fields and ensuring compatibility with other plugins, as each handles and stores meta information differently. Additionally, updating checkout fields on the frontend is more complex due to the React-based checkout, making it harder to modify field data compared to a vanilla JS or jQuery-based approach.

Having a PHP or even a JS hook to override the customer assigned to the checkout order would simplify this process while maintaining backward compatibility. As @webdados mentioned, this functionality already exists in the classic checkout.

**Comment by pierorocca** [comment]
Hey @s3rgiosan could you help me understand the use case/business case that requires this kind of flow? On the surface this seems like a concierge/impersonation type flow?

**Comment by webdados** [comment]
@pierorocca It's indeed a "concierge" flow, [as I mentioned above](https://github.com/woocommerce/woocommerce/issues/44530#issuecomment-2730108540). @s3rgiosan is talking about the same project. Not an "impersonation" flow, tho.

**Comment by pierorocca** [comment]
@webdados thanks for the extra info. Does this mean you're also taking the credit card info over the phone and keying it in on behalf of the shopper?

**Comment by webdados** [comment]
😂 @pierorocca Not me, no. I'm not taking credit cards over the phone.

This is a plugin that is used by 900+ shops around the world. What each shop does, it's their problem, not mine.
For years, hotels and travel agencies have been doing this :-)

The plugin does have a fake payment gateway called "Request payment by email", that only the shop owner/manager sees and can/should choose on checkout, which will send an email to the customer with the payment link on WooCommerce, so that the customer himself can pay for the order with whatever payment method is available. This is what we recommend the shop owner to do in case the customer wants to use a credit card.

<img width="1076" alt="Image" src="https://github.com/user-attachments/assets/513e8246-098f-4725-9244-25c2d30083b6" />

Also, there are other payment methods around the world besides credit card, which pose no security issues that requesting the card number over the phone pose. For example in Portugal we have methods like Multibanco (pay with a reference at the ATM or home banking service), or MB WAY which is an app tied to your phone number and bank account (the customer just provides the mobile number and waits for an push notification to approve the payment on the app, after the order is submitted). I'm the author of the plugin that adds those methods to WooCommerce: https://wordpress.org/plugins/multibanco-ifthen-software-gateway-for-woocommerce/

**Comment by webdados** [comment]
Anyway, we're just asking to have the same hook on the Blocks Checkout that exists for years on the Classic Checkout.

**Comment by pierorocca** [comment]
https://ptwooplugins.com/product/shop-as-client-for-woocommerce-pro-add-on/

Thanks Marco. I have some issues with this model, particularly where credit card numbers and CVV could be captured over the phone and entered on behalf of the user. This has PCI and risk implications. I understand Credit Card may or may not be used but you don't have any control over that once the plugin is installed. This creates a control gap for payments gateways.

As I've noted elsewhere on GH, it's an unsupported workaround to controls a payment gateway might have in place. For example on WooPayments we don't support phone orders for credit card aka Virtual Terminal. This is an intentional business decision. This plugin, correct me if I'm worng, creates a workaround for credit/debit cards.

The other issue I have, particularly for Europe and related to GDPR and privacy, the shopper isn't the one accepting the T&Cs and privacy policy. It's the store admin doing it on their behalf. How does the plugin ensure shopper consent is captured by the actual buyer?

I like the creativity of the app but I question how legal and safe it is and why it's a better option that using the back-end order flow. An FYI that at some point I likely plan on replacing the Pay for Order flow with the standard checkout flow. It would be able to elegantly handle manually created orders and collect payments and policy consent in a safe, compliant way.

Thanks for the education.

**Comment by webdados** [comment]
Thanks Piero. I understand your concerns, but I disagree with them.

The back-end order flow does not calculate shipping, or benefits from any of the discount rules you might have in place on the frontend, or supports any of the countless features thousands of plugins might have implemented on the frontend. That's what we are solving here.

The plugin does not create an workaround for anything. It allows a shop owner to get phone or email orders. That's it. (It was actually developed out of my own necessity, as I do run 5 shops in Portugal, none of those accept credit card, as people prefer the methods I mentioned above, as they are far more secure).

It's not the plugin that is creating a legal problem. Anyone can use WooCommerce to sell ilegal items, and that's not a reason to stop developing WooCommerce, right?
(I thought of several other analogies, but I'm worried they might be misunderstood, so I'll keep them to myself 😂)

There are [several options](https://www.google.com/search?q=woocommerce+phone+orders) to do this with WooCommerce. This is just another one.

I have a big customer that found a new use-case for it: they have sales reps visiting customers (retail shops) all across the US, and they are using WooCommerce and my plugin to accept the orders on the go, and assigning them to the customer user on the shop. There's no payment involved on that moment. And they use the meta stored on the order (handler user id) to calculate commission for the sales rep. How good of a problem solver is this?

I don't think us asking to create the same filter on the blocks checkout that already exists on the classic checkout should turn into a discussion of what my plugin does is legal or not, because it's just a tool, like WooCommerce is. It's up to the shop owners to use these tools in accordance to their local laws.

This ticket is a mere technical request for WooCommerce to do something it already does and make the checkouts more coherent, nothing more than that.

**Comment by pierorocca** [comment]
> The back-end order flow does not calculate shipping, or benefits from any of the discount rules you might have in place on the frontend, or supports any of the countless features thousands of plugins might have implemented on the frontend.

This is correct and it does suck, hence why I'm proposing to revamp the flow so manual orders trigger a standard checkout flow where shipping, discounts, and all the 3P plugin goodness can execute. I'm not a fan of the existing pay for order flow as in my mind, it's extremely error prone and functionally lacking. It's somewhere between an invoice and a standard checkout and does neither well. This will likely make plugins like yours unnecessary in the future, for the benefit of merchants.

> The plugin does not create an workaround for anything. It allows a shop owner to get phone or email orders. That's it. (It was actually developed out of my own necessity, as I do run 5 shops in Portugal, none of those accept credit card, as people prefer the methods I mentioned above, as they are far more secure).

Let's run through the scenario of a store that adds the plugin and they're using a Stripe gateway with credit card enabled. I totally get Multibanco (which WooPayments is about to launch) and other non-card payment methods are more secure. But I don't control what a site does or doesn't support so I have to assume the worst case that Credit/Debit cards could be offered as a payment method and a store uses it as a workaround for phone orders.

When a shopper is on checkout, the device they're using can be fingerprinted and other signals sent as part of fraud and risk controls. If that shopper isn't present on checkout and someone else keys in the information on their behalf, that digital metadata is not available for fraud and risk screening. That creates risk for Stripe who financially underwrites the merchant. Additionally since the credit card number and CVV are taken over the phone, there's less control over how the information is accessed, stored, shared, destroyed and it makes it harder to be PCI compliant. Again I understand that you professionally don't use credit card. It's the potential for someone to use it with a credit card and lack of control I'd have to prevent that is the concern.

Another side-effect by having a concierge shop on the front-end it skews data on conversion, analytics, attribution, etc.

> This ticket is a mere technical request for WooCommerce to do something it already does

I understand that this is something classic checkout "supports", unfortunately now that I have a clearer understanding it's not something I can propagate support for due to the implications/risk it creates for payments gateways, specifically card payments. On Blocks, the way forward here is to re-think the manual order creation process so that we get the best of both worlds - stores can create orders on behalf of shoppers/clients AND buyers can pay through a fully functional standard checkout that properly calculates shipping, allows for gift cards and loyalty redemption, order bumps, upsells/cross-sells, etc. 

I'm sorry that's not the answer you're looking for here and this is an area I have to be cautious of the downstream implications.

**Comment by fabiomsnunes** [comment]
@pierorocca, how is that any different from a sales rep checking out on behalf of the customer using incognito?

I understand and share your concerns, but you can’t just assume it will be used for illegal reasons. The sales commission example that @webdados shared is a perfect, completely legal use case

**Comment by webdados** [comment]
@pierorocca We can easily make payment methods like Stripe, WooPayments, Paypal, and others unavailable for the Shop Owner / Manager when they are in "Shop as Client" mode and force them to use the "Request Payment by Email" method, as well as make more visible warnings about not using similar payment methods as a shop owner, but I guess it wouldn't change anything regarding the original request and your decision about it.

In the end, it's Automattic/WooCommerce decision to add or not add the filter to the new checkout, and while I share your security concerns, I think the target here is wrong. It's just a tool, like the core WooCommerce is (which can also be used to do bad as is).

**Comment by pierorocca** [comment]
> In the end, it's Automattic/WooCommerce decision to add or not add the filter to the new checkout, and while I share your security concerns, I think the target here is wrong. It's just a tool, like the core WooCommerce is (which can also be used to do bad as is).

I hear you Marco but I'm stuck here as the gateway, the one owning the financial risk and compliance with PCI DSS, doesn't retain control of that decision and other developers using the filter may not be as thorough as you. It's a vulnerability.

 Let's come back to this please when I start thinking about what the future of the Pay for Order flow should be. This on-behalf will be a use case I'll add so we can think about things like either an enhanced back-end order experience or building out a structured way on the front end for access to a like filter, but if the filter is used, payments are deferred until the shopper is present. Does that sound like a viable plan?

**Comment by webdados** [comment]
I guess it's the only plan for now :-)

We'll also improve the communication on our plugin about the usage of payment methods, and add further emphasis on the usage of the "request payment by email" anytime a customer wants to pay with a payment method such as PayPal or Credit Card.

**Comment by pierorocca** [comment]
Thanks @webdados. We can also look at a class of **_card_** transactions called [Merchant Initiated Transactions (MIT)](https://docs.stripe.com/payments/cits-and-mits?locale=en-GB#merchant-initiated-transactions-(mit)).  This allows for merchants, with prior authorization, to place orders on behalf of the shopper based on prior instructions/terms that have been captured. This might allow us in certain situations to even further streamline the flow, though I'd like to look at the total list of variations:

1. Concierge completes the full order including payment, IF the shopper has a card on file and has accepted the terms
2. Concierge completes the order excluding payment, shopper is notified, and shopper completes the payment
 
Q1: Do you envision a scenario where the shopper is not yet a registered user of the store?
Q2: In your current plugin, I assume the "Concierge" is role restricted? i.e. bots couldn't for example start placing orders or scrape emails and customer info?

**Comment by webdados** [comment]
Even if the shopper has a card on file, the concierge cannot use it at this moment in our plugin, because the concierge is logged as himself, not the user account.

MIT is (I think) what WooCommerce Subscriptions uses to charge renewals. We have it on our own plugins shop.
I guess we could look into how to implement something like this:

- When the shop manager selects the user on our plugin, we look internally for stored cards and inform the shop manager that it exists (the only way for the card to exist on file is the customer to have entered it logged-in to his own account - we just need to figure out the "authorization to use in later orders" part)
- Have another custom payment gateway called something like "Pay with customer card on file" and maybe list them so that the shop manager can ask the customer which to use, if there's more than one
- Initiate the payment on Stripe in a similar way WooCommerce Subscriptions does for each renewal, in the background, upon order saving and assigning to the customer by our plugin

Unrelated to MIT, another thing we can consider doing on the plugin, is to disable by default PayPal / Stripe / WooPayments card / ... whenever the user is "Shopping as a client" and not for himself, thus reducing drastically the chance for any plugin user to request card details over the phone. The problem with this approach is that we have no idea how many credit card payment plugins exists out there and it's virtually impossible to block them all.


Q1: Yes, this is a very common scenario and our plugin allows the creation of the user account at that exact moment.

Q2: Yes, by default it's only available for logged-in Administrators and Shop Managers (WooCommerce created role). There's a [developer filter](https://gist.github.com/webdados/0f1a8e5ca2cd98276a9ae73918e9b842) that allows shop owners to further authorize the usage of this functionality to other WordPress user roles (not logged-out users, ever). For example, on the use-case I mentioned some posts ago, the shop has a custom "Sales rep" user role that has no access to any admin functionality or order management, but can use his profile to enter orders for the customers they're visiting on the road, because they used the filter to allow that user role to use the "Shop as Client" functionality. Let me reinforce the fact that the logged-in user will never have access to the customer cards on file or tokens because they are logged-in as themselves.

**Comment by pierorocca** [comment]
> MIT is (I think) what WooCommerce Subscriptions uses to charge renewals.

Yes subscriptions, and any type of recurring billing is considered to be a MIT type of transaction. I really like this visual from Visa that shows the categories of use cases:

![Image](https://github.com/user-attachments/assets/8ebb1da2-8673-4d3d-8ff0-f6775a1eaa70)

> Unrelated to MIT, another thing we can consider doing on the plugin, is to disable by default PayPal / Stripe / WooPayments card / ... whenever the user is "Shopping as a client" and not for himself, thus reducing drastically the chance for any plugin user to request card details over the phone.

Yes this was my thinking in my "deferred payment until when the shopper is present" comment. This way the experience is orchestrated and the vulnerabilities controlled. I think we're on the same page.

In the back of my mind I'm thinking about grocery use cases where I as a shopper place an order, then a store buyer tries to secure the products I've asked for but often they have to make a substitution or remove an item. The total and taxes have to get updated on the order, I have a window of time to approve, and then the order is finalized with the card on file.

Q1: Thanks for confirming. How do you handle presentment of and acceptance of the privacy policy in that case? I'm struggling with this a bit because once the email and name is stored, it's stored and digital consent isn't yet obtained.

Q2: Makes sense thank you.

> Let me reinforce the fact that the logged-in user will never have access to the customer cards on file or tokens because they are logged-in as themselves

Yes absolutely. What I'm worried about, based on other requests I've seen elsewhere, is opening up the card fields in a way that allows a customer service agent for example to type in the card number that's obtained over the phone. That presents PCI DSS challenges because it's hard to manage what that person does with the number and the security code and other details about the person. That model also requires different solutions, different underwriting and ToS, and different risk management than a standard metadata rich web transaction. Also a phone based transaction has to be tagged in a certain way for the networks to know it was phone based and it can change who owns the chargeback liability.

**Comment by mikejolley** [comment]
Responding after seeing https://github.com/woocommerce/woocommerce/pull/65780 which attempted to add the filter to Store API.

The use case is legitimate and the parity gap with legacy is valid. The mechanism is the issue.

The classic filter reassigns the order's customer part-way through checkout. Core and the Store API assume the order customer is the WC session customer, and the logged-in user when there is one. Overriding it mid-flow can things break later:

- Saved cards / tokens belong to whoever's logged in (the operator), not the customer, so they can end up on the wrong account.
- The order confirmation page checks the order customer against the logged-in user and fails after the swap.
- Customer meta, downloads, tax/exemption are read from the session during checkout, not re-derived afterwards.

There are existing Store API available mid-flow hooks which make this possible without implementing `woocommerce_checkout_customer_id` (`woocommerce_store_api_checkout_update_order_from_request`), so this is not a gap, but they have the same problem, so they're not the answer either.

The direction we'd want is to switch the customer at the session level before checkout runs: operator selects the customer, the session moves into an "ordering on behalf of" mode (cookie plus customer session handler), and the session customer is that customer throughout. Its more work upfront, but it keeps each account's data isolated.

I think that given there are alternatives already (`woocommerce_store_api_checkout_update_order_from_request`), the gap has existed for quite some time, and the mid-flow hooks are not ideal, we should close this request as won't-fix.

cc @senadir

**Comment by webdados** [comment]
Understood @mikejolley - Thanks for looking into this again.
